widget/ErrorInstancesTable.tsx

```diff
@@ -3,37 +3,101 @@
 
 import React, { Component } from "react";
 import ReactDOM from "react-dom";
-import * as d3 from "d3";
+import {
+  DetailsList,
+  DetailsListLayoutMode,
+  Selection,
+  SelectionMode,
+  IColumn
+} from "office-ui-fabric-react/lib/DetailsList";
+import { DefaultButton } from 'office-ui-fabric-react';
+import { Fabric } from "office-ui-fabric-react/lib/Fabric";
+import { apiBaseUrl } from "./api.ts";
 
 
 type ErrorInstancesTableState = {
-  data: any
+  selecedDataPoint: any,
+  page: number
 }
 
 type ErrorInstancesTableProps = {
-  data: any
+  selectedDataPoint: any,
+  pageSize?: number
 }
 
 class ErrorInstancesTable extends Component<ErrorInstancesTableProps, ErrorInstancesTableState> {
+
+  public static defaultProps = {
+      pageSize: 5
+  };
+
   constructor(props) {
     super(props);
-
     this.state = {
-      data: this.props.data
-    };
+      selecedDataPoint: null,
+      page: 0
+    }
   }
 
   componentWillReceiveProps(nextProps) {
     this.setState({
-      data: nextProps.data
+      selecedDataPoint: nextProps.selecedDataPoint
     });
   }
 
   render() {
+
+    if (this.props.selectedDataPoint == null) {
+      return (
+        <React.Fragment />
+      );
+    }
+    var columns = [
+      {
+        key: 'instanceId',
+        name: 'Instance',
+        fieldName: 'instance_id',
+        minWidth: 100,
+        maxWidth: 100,
+        isResizable: false ,
+        onRender: (instance) => {
+          return (<img src={`${apiBaseUrl}/api/v1/instance_data/${instance.instance_id}`} />);
+        }
+      },
+      { key: 'h1Prediction', name: 'h1 Prediction', fieldName: 'h1_prediction', minWidth: 100, maxWidth: 100, isResizable: false },
+      { key: 'h2Prediction', name: 'h2 Prediction', fieldName: 'h2_prediction', minWidth: 100, maxWidth: 100, isResizable: false },
+      { key: 'groundTruth', name: 'Ground Truth', fieldName: 'ground_truth', minWidth: 100, maxWidth: 100, isResizable: false },
+    ];
+
+    var items = [];
+    for(var i=(this.state.page * this.props.pageSize); i < (this.state.page * this.props.pageSize) + this.props.pageSize; i++) {
```

Comment:
I think you need

`i < Math.min((this.state.page * this.props.pageSize) + this.props.pageSize, this.props.selectedDataPoint.error_instances.length)`

to avoid walking off the error_instances array

Reply:
Good catch. Thank you.

Fixed @Nking92